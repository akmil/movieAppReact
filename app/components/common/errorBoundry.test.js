import React from 'react';
import renderer from 'react-test-renderer';
import ErrorBoundary from './errorBoundry';
import { shallow, mount } from 'enzyme';


const TestComponent = ({ throwError }) => {
  if (throwError) {
    throw new Error('Some exception');
  }

  return <div>Everything looks fine</div>;
};

describe('ErrorBoundary', () => {
  beforeEach(() => {
    // wrapper.setState({ hasError: true });
    // jest.spyOn(window._virtualConsole, 'emit').mockImplementation(() => false);
  });

  it('should render without errors ErrorBoundary', () => {
    const wrapper = shallow(<ErrorBoundary><TestComponent/></ErrorBoundary>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with hasError ErrorBoundary', () => {
    const wrapper = shallow(<ErrorBoundary><TestComponent/></ErrorBoundary>);

    wrapper.setState({ hasError: true });
    expect(wrapper.text()).toEqual('Something went wrong.');
    expect(wrapper.state().hasError).toBe(true);
  });

  it('should catch error', () => {
    const wrapper = mount(<ErrorBoundary><TestComponent throwError={true}/></ErrorBoundary>);
    expect(wrapper.state().hasError).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
