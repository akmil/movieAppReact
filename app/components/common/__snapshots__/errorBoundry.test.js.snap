// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should catch error 1`] = `
<ErrorBoundary>
  <h1>
    Something went wrong.
  </h1>
</ErrorBoundary>
`;

exports[`ErrorBoundary should render without errors ErrorBoundary 1`] = `<TestComponent />`;
